src/EFCore.PG/Extensions/NpgsqlPropertyExtensions.cs

```diff
@@ -151,6 +151,12 @@ public static NpgsqlValueGenerationStrategy GetValueGenerationStrategy([NotNull]
                 return (NpgsqlValueGenerationStrategy)annotation;
             }
 
+            var sharedTablePrincipalPrimaryKeyProperty = property.FindSharedTableRootPrimaryKeyProperty();
+            if (sharedTablePrincipalPrimaryKeyProperty != null)
+            {
+                return sharedTablePrincipalPrimaryKeyProperty.GetValueGenerationStrategy();
+            }
```

Comment:
Maybe rewrite with pattern matching?
```suggestion
            if (property.FindSharedTableRootPrimaryKeyProperty() is {} rootPrimaryKeyProperty)
                return rootPrimaryKeyProperty.GetValueGenerationStrategy();
```

Reply:
I'm probably going to change this to exclude SequenceHiLo, depending on the outcome in https://github.com/dotnet/efcore/pull/20451#issuecomment-605545100. Other than that sure...

Reply:
Changed to exclude SequenceHiLo

Reply:
Also changed to use pattern matching